src/menu/src/MenuItem.js

```diff
@@ -55,6 +56,28 @@ const MenuItem = memo(
       internalStyles
     )
 
+    let iconColor = intent === 'none' ? 'default' : intent
+
+    if(disabled) {
+      iconColor = 'disabled'
+    }
+
+    const textColor = intent
+
+    const secondaryTextColor = disabled ? textColor : 'muted'
+
+    const disabledProps = useMemo(() => {
+      return disabled ? {
+        cursor: 'not-allowed',
+        disabled: true,
+        onClick: null,
+        onKeyPress: null,
```

Comment:
These should probably be `undefined`? or maybe handled in the definitions of `handleClick` and `handleKeyPress`? (I realize these changes aren't from your PR, but revisiting them now).